lib/engine/step/tracker.rb

```diff
@@ -26,6 +27,7 @@ def get_tile_lay(entity)
         action[:lay] = !@upgraded if action[:lay] == :not_if_upgraded
         action[:upgrade] = !@upgraded if action[:upgrade] == :not_if_upgraded
         action[:cost] = action[:cost] || 0
+        action[:upgrade_same_hex] = true if action[:upgrade_same_hex].nil?
```

Comment:
unless action[:upgrade}

Reply:
that doesn't work since action[:upgrade_same_hex] unless i flip the tense to cannot_reuse_same_hex

Reply:
ok no problem this is fine then

Reply:
I'd already done it, which would you prefer?